### PR TITLE
Forward leave reason of disconnected events

### PR DESCRIPTION
**Pull request by lukasIO**
client implementation for https://github.com/livekit/protocol/pull/139

### CODE REVIEW

File: src/room/Room.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import {
 import {
   ConnectionQualityUpdate,
   JoinResponse,
+  LeaveRequest_LeaveReason,
```
Comment (davidzhao) [inline src/room/Room.ts:18]: what do you think about exporting a type-defed version of this? i.e. `type DisconnectedReason = LeaveRequest_LeaveReason`, it seems a bit more friendly for users to import.

Comment (lukasIO) [reply src/room/Room.ts:18]: sounds good! was missing the export anyways...

Comment (davidzhao) [reply src/room/Room.ts:18]: or.. maybe I should just name that DisconnectReason in the proto? 🤔 wdyt?

Comment (lukasIO) [reply src/room/Room.ts:18]: renaming in the proto seems slightly cleaner, but no strong opinion either way.

Comment (davidzhao) [reply src/room/Room.ts:18]: yeah, you are right.. it'll be consistent across platforms.. I'll do that and then merge.

Comment (davidzhao) [reply src/room/Room.ts:18]: yeah, you are right.. it'll be consistent across platforms. will update.

**Review by davidzhao — APPROVED** [review]
lg!

**Pull request merged**